webapp/platform/client/src/websocket.ts

```diff
@@ -377,11 +419,12 @@ export default class WebSocketClient {
             this.responseCallbacks[msg.seq] = responseCallback;
         }
 
+        // Only try to send the message if the websocket is open.
+        // If the websocket is closed here, we will drop the message,
```

Comment:
This comment is unclear to me.  Messages will be dropped if a reconnect is in progress?

Reply:
The last half of this comment isn't helpful, so I dropped it: [332ccd9](https://github.com/mattermost/mattermost/pull/30135/commits/332ccd904843064991d69bf22b452787411e4898)

If the websocket is closed, we always drop the message. That was the behavior before my changes.

What has changed is that we don't immediately try to re-connect the websocket if it's closed when sending. That responsibility (trying to ensure the websocket is open) is handled elsewhere 🙂

Reply:
Understood. From a correctness perspective, dropping those messages is bad, versus having a small queue that gets drained when the websocket is reconnected.  I think this sorta works today because messages from client tend to be non-critical (e.g. typing messages).  Agreed this is existing behavior and not in scope to address in this PR.

Reply:
Created a Jira task to address this here: https://mattermost.atlassian.net/browse/MM-63221